pkg/controller/volume/expand/pvc_populator.go

```diff
@@ -107,6 +107,13 @@ func (populator *pvcPopulator) Sync() {
 			eventType := v1.EventTypeNormal
 			if err != nil {
 				eventType = v1.EventTypeWarning
+			} else {
+				// Do not log and record the event when we can not found the plugin
+				// and the StorageClassName is not set at the same time.
+				// See https://github.com/kubernetes/kubernetes/issues/77404
+				if len(pv.Spec.StorageClassName) == 0 {
```

Comment:
On this place we hit race between PV controller binding the PVC and resize controller checking the PVC. The PVC is probably not bound yet, so its `pvc.status.capacity` is zero and thus less than `pvc.spec.resources`.

Should there be check if the PVC is bound instead?

@gnufied, @mlmhl PTAL

Reply:
> On this place we hit race between PV controller binding the PVC and resize controller checking the PVC. The PVC is probably not bound yet, so its `pvc.status.capacity` is zero and thus less than `pvc.spec.resources`.
> 
> Should there be check if the PVC is bound instead?
> 
> @gnufied, @mlmhl PTAL

Seems in this loop, we would found the PV according to PVC spec first.So if it not Bound,  we could would not find the PV, then it would not continue?

Reply:
As a interim measure until https://github.com/kubernetes/kubernetes/pull/75386 lands, we can move checking for bound and storageclass into a util function inside `pkg/volume/util/resize_util.go` and perform same check as https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/expand/cache/volume_resize_map.go#L99 , so as we don't create unnecessary events.

Reply:
Also in which case we should remove checks we are performing on PVC inside `AddPVCUpdate` since we are already doing that prior to adding PVC to reconciler loop.